core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/Stream.scala

```diff
@@ -0,0 +1,432 @@
+package com.lynbrookrobotics.potassium.streams
+
+import java.util.concurrent.Semaphore
+
+import com.lynbrookrobotics.potassium.clock.Clock
+import squants.Quantity
+import squants.time.{Time, TimeDerivative, TimeIntegral}
+
+import scala.collection.immutable.Queue
+import scala.collection.mutable
+import scala.ref.WeakReference
+
+import com.lynbrookrobotics.potassium.Platform
+
+abstract class Stream[T] { self =>
+  final type Value = T
+
+  val expectedPeriodicity: ExpectedPeriodicity
+
+  private[this] val listeners = mutable.Queue.empty[T => Unit]
+
+  protected def publishValue(value: T): Unit = {
+    synchronized {
```

Comment:
Why do we need this to be synchronized? I don't imagine there'll be multiple threads sharing a common Steam and calling publishValue